### ISSUE TITLE
Oauth2 access

### ISSUE THREAD

**Issue by SaschaStenger** [DocNow/twarc#302]
Hello there. I wanted to know if twarc does support `Oauth2` access tokens for read only search with bigger limits, then the standard `Oauth1`. Because so far i couldn't find anything about it in the documentation about getting the token, when using twarc as a library.

**Comment by edsu** [comment]
That's a great question @SaschaStenger. At the moment it does not, but it shouldn't be difficult to add. I'll look into it.

**Comment by SaschaStenger** [comment]
That's great to hear, thank you. I'll have a look out, in case you update the connection type.

**Comment by rmcrowley2000** [comment]
It looks like this is essentially a duplicate of #246.

This is definitely an important issue though, at least for those using the Search API.  It's a difference of 180 queries per 15 minutes per **user** versus 450 per 15 minutes per **app** (at multiple apps per user).  As a researcher, this is a big difference when scraping for multiple projects.

**Comment by edsu** [comment]
Good point @rmcrowley2000. I'll leave it open as additional impetus for actually doing something about it!

**Comment by edsu** [comment]
Ok, you should be able to use --app_auth on the command line to control whether App Auth is being used. This is super useful in cases when you know the rate limits are better for App Auth (search) but not so useful in cases where it isn't (hydration for example).

**Comment by edsu** [comment]
Just was just released as part of v1.8.0 to PyPI.
